### PR TITLE
Support consul tokens

### PR DESCRIPTION
**Pull request by akissa**
* Adds support for consul tokens
* Adds support for consul 1.x.x which enforces HTTP verbs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ The `consul-register` option is keyval based, and it takes the following mendato
 
 Optional key options:
 
+* `token` : the consul token to use.
```
Comment (ultrabug) [inline README.md]: please sort the option alphabetically, just like you rightfully did on the rest of the code :)

**Comment by ultrabug** [comment]
Thanks a lot for your contribution @akissa !

A small comment to take care of please. I suppose you tested it right?

**Comment by akissa** [comment]
@ultrabug Yes i have tested and it is working okay for me so far not run into any issues running it in semi production.

**Comment by akissa** [comment]
@ultrabug ping

**Comment by ultrabug** [comment]
Sorry for my late reply and poor reactivity @akissa 

Thanks again for your contribution!

**Pull request merged**